src/anbox/cmds/session_manager.cpp

```diff
@@ -131,19 +131,21 @@ anbox::cmds::SessionManager::SessionManager()
                       use_system_dbus_));
 
   action([this](const cli::Command::Context &) {
-    auto trap = core::posix::trap_signals_for_process(
+    core::posix::SignalTrap* trap = core::posix::trap_signals_for_process(
         {core::posix::Signal::sig_term, core::posix::Signal::sig_int});
     trap->signal_raised().connect([trap](const core::posix::Signal &signal) {
       INFO("Signal %i received. Good night.", static_cast<int>(signal));
       trap->stop();
     });
 
     if (standalone_ && !experimental_) {
+      delete trap;
```

Comment:
Why are we switching from a std::shared_ptr to a raw pointer?

Reply:
This format is changed because the cycling-reference of shared_ptr is checked in valgrind, I replace it with pointer with explicit new and delete. This time the valgrind doesn't complain it.

Reply:
cycling-reference of shared_ptr causes memory leak.

Reply:
I've reverted this commit.